shared_model/interfaces/utils/specified_visitor.hpp

```diff
@@ -26,21 +26,30 @@ namespace shared_model {
   namespace interface {
     template <typename Type>
     class SpecifiedVisitor
-        : public boost::static_visitor<
-              boost::optional<shared_model::detail::PolymorphicWrapper<Type>>> {
+        : public boost::static_visitor<boost::optional<const Type &>> {
      private:
-      using Y = shared_model::detail::PolymorphicWrapper<Type>;
+      using Y = shared_model::detail::PolymorphicWrapper<
+          std::remove_cv_t<std::remove_reference_t<Type>>>;
```

Comment:
maybe remove only const?

Reply:
why remove volitile?

Reply:
This using will be removed after polymorphic wrapper is removed, and now polymorphic wrapper is only used with non-const non-reference types.
@l4l Since polymorphic wrapper stores shared pointer, having a pointer to a reference seems fragile.
@kamilsa yes, `std::remove_const` can be used here